File: data_structures/stack/cpp/stack.cpp
#include <iostream>
using namespace std;

class stack{
  public:
    char arr[10];
    int top;

    stack(){
      top = -1;
    }

    void push(char x){
      if(top<10){
        top+=1;
        arr[top] = x;
      }
      else{
        cout<<"Stack is full\n";
      }
    }

    char pop(){
      if(isEmpty()){
        cout<<"Stack is Empty\n";
      }
      char a = arr[top];
      top-=1;
      return a;
    }

    char topel(){
      return arr[top];
    }

    bool isEmpty(){
      if(top==-1){
        return true;
      }
      return false;
    }

    void print(){
      for(int i=0; i<=top; i++){
        cout<<arr[i]<<" ";
      }
      cout<<"\n";
    }
};

File: data_structures/stack/csharp/Stack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataStructures
{
    class Stack
    {
        private List<Object> items = new List<Object>();

        public Object pop()
        {
            if (items.Count > 0)
            {
                int lastPos = items.Count - 1;
                Object temp = items[lastPos];
                items.RemoveAt(lastPos);
                return temp;
            }
            else
            {
                throw new Exception("Underflow: Stack is empty");
            }
        }

        public void push(Object newItem)
        {
            items.Add(newItem);
        }


        public bool isEmpty()
        {
            return (items.Count == 0);
        }

    }
}
